blip-00xx.md

```diff
@@ -0,0 +1,220 @@
+- SHOULD sign the invoice with a private key that is not the same as their 
+  public node ID and should not set the destination node (`n` field).
+
+### Size Restrictions
+
+#### `data_length` Limit
+
+In order to conform to any existing BOLT 11 invoice parser, each new tagged 
+field must use the `data_length` encoding defined there. This means that the 
+maximum size of any _single_ encoded blinded path is 639 bytes. 
+
+What follows is a rough estimation of the maximum number of hops we can include 
+in a single blinded path. It assumes that each hop's cipher text is the same 
+length. 
+
+##### Cipher Text Size Estimation
+
+First, a rough estimation of the average cipher text length is required. A 
+forwarding node in a blinded path will receive a cipher text payload containing
+the following data:
+
+- `padding`: optional
+- `short_channel_id` of 8 bytes
+- `payment_relay`: 
+    * 2 byte `cltv_expiry_delta`
+    * 4 byte `fee_proportional_millionths`
+    * 4 byte `fee_base_msat`
+- `payment_constraints`:
+    * 4 byte `max_cltv_expiry`
+    * 8 byte `htlc_minimum_msat`
+- `allowed_features`: optional
+
+If we assume that the `allowed_features` vector is not set, then this comes to 
+a total of 30 mandatory bytes. 
+
+For the recipient node, it will receive a cipher text payload containing:
+
+- `padding`: optional
+- `path_id`: let's assume that this is 32 bytes like the existing payment 
+   address.
+- `payment_constraints`:
+    * 4 byte `max_cltv_expiry`
+    * 8 byte `htlc_minimum_msat`
+
+This comes to a total of 44 bytes for the recipient's cipher text. 
+
+The padding field should be used by recipients to pad cipher text blobs so that 
+all are the same size. Since the calculated recipient cipher text blob size (44) 
+is larger than that of the forwarding nodes (30), we can assume that all the
+cipher text blobs will have a size of around 44 bytes. 
+
+##### `blinded_hop` Size Estimation
+
+The total number of bytes required for a single `blinded_hop` is:
+
+    = 33+2+len(cipher_text)
+    = 35+len(cipher_text)
+
+If we use the estimated `cipher_text` size of 44 bytes, then this comes to 79 
+bytes for a single `blinded_hop`.
+
+##### `blinded_payinfo` Size Estimation
+
+The total number of bytes required for the encoding of a single 
+`blinded_payinfo` entry is:
+
+    = 4+4+2+8+8+2+len(features)+33+1+num_hops(len(blinded_hop))
+    = 68+len(features)+num_hops(len(blinded_hop))
+
+If we take the estimate of 79 bytes per `blinded_hop` and if we assume an empty 
+feature vector then this comes to:
+
+    = 68+num_hops(79)
+
+The maximum number of hops in a single blinded path can then be calculated to 
+be:
+
+    639 = 68+num_hops(79) 
+    num_hops = 7
+
+#### QR code limit
+
+Another soft maximum value to keep in mind is the maximum number of bytes that
+can fit into a [QR code][qr] which is 2,953 bytes. This is a soft maximum 
+because this only applies if the invoice is in fact being transmitted via QR 
```

Comment:
Question:  How is this limit being used in this proposal? Is it enforced in anyway to limit the data which can be included?

Given that this constraint is applicable only if the invoice is being displayed as a QR code, would it make sense to add an explicit logic to indicate modality of invoice transmission and enforce the limits based on that?

Reply:
> How is this limit being used in this proposal?

It's not. It's just informative. This is purely for implementation help.

> add an explicit logic to indicate modality of invoice transmission and enforce the limits based on that

That is really up to the implementation I'd say